includes/utilities.inc

```diff
@@ -310,51 +309,37 @@ EOQ;
  */
 function islandora_newspaper_set_mods_date_issued(AbstractDatastream $datastream, DateTime $date) {
   $out = FALSE;
-  $file = file_create_filename("{$datastream->parent->id}_{$datastream->id}.xml", 'temporary://');
-  $datastream->getContent($file);
-  @$doc = simplexml_load_file($file);
-  if ($doc) {
-    $doc->registerXPathNamespace('ns', 'http://www.loc.gov/mods/v3');
+  $dom = new DOMDocument();
+  $dom->loadXML($datastream->content);
+  $xpath = new DomXPath($dom);
+  if ($dom) {
+    $xpath->registerNamespace('ns', 'http://www.loc.gov/mods/v3');
     // Assumes the canonical date issued exists in the first mods document under
     // origin info and is not specified as a point, additional logic could be
     // added to process different encodings.
-    $parent = FALSE;
-    $dates = $doc->xpath('//ns:mods[1]/ns:originInfo/ns:dateIssued[not(@point)][1]');
-    if (is_array($dates) && count($dates) > 0) {
-      $removal = array();
-      // XX: Because you can't remove elements in a foreach loop
-      // we collect them. Then we can add the new one and remove
-      // the old.
+    $dates = $xpath->query('//ns:mods/ns:originInfo/ns:dateIssued[not(@point)]');
+    if ($dates->length > 0) {
```

Comment:
Ah, I guess we can remove this `if` check as well, no? Since `$dates` is properly `Traversable` and all